lightrag/lightrag/components/model_client/lite_client.py

```diff
@@ -0,0 +1,175 @@
+import os
+from typing import Dict, Optional, Any, List
+import backoff
+import logging
+
+from lightrag.utils.lazy_import import safe_import, OptionalPackages
+from lightrag.core.model_client import ModelClient
+from lightrag.core.types import ModelType, EmbedderOutput, Embedding
+
+# Conditional import of litellm
+litellm = safe_import(OptionalPackages.LITE.value[0], OptionalPackages.LITE.value[1])
+
+from litellm import (
+    get_supported_openai_params,
+)
+from litellm.exceptions import (
+    Timeout,
+    InternalServerError,
+    APIConnectionError,
+    RateLimitError,
+)
+
+log = logging.getLogger(__name__)
+
+
+class LiteClient(ModelClient):
+    """A component wrapper for the LiteLLM API client.
+
+    Visit https://litelms.com/docs/ for more api details.
+    Check https://litelms.com/docs/models for the available models.
+
+    Tested LiteLLM models: 07/19/2024
+    - mixtral-8x7b
+    - gemma-7b-it
+    """
+
+    def __init__(
+        self,
+        model: Optional[str] = None,
+        api_key: Optional[str] = None,
+        base_url: Optional[str] = None,
+        api_version: Optional[str] = None,
+        timeout: Optional[int] = 600,
+        drop_params: bool = False,
+    ):
+        super().__init__()
+        self.model = model
+        self._api_key = api_key or os.environ.get("GROQ_API_KEY")
+        self.base_url = base_url
+        self.api_version = api_version
+        self.timeout = timeout
+        self.drop_params = drop_params
+        self.sync_client = self.init_sync_client()
+        self.async_client = None  # only initialize if the async call is called
+
+    def init_sync_client(self) -> Any:
+        import litellm
+
+        validate = litellm.validate_environment(model=self.model)
+        if not validate["keys_in_environment"]:
+            ApiError = validate["missing_keys"]
+            raise ValueError(
+                f"Environment variable {ApiError} required for LiteLM model {self.model} must be set"
+            )
+        return litellm
+
+    def init_async_client(self) -> Any:
+        import litellm
+
+        validate = litellm.validate_environment(self.model)
+        if not validate["keys_in_environment"]:
+            ApiError = validate["missing_keys"]
+            raise ValueError(f"Environment variable {ApiError} must be set")

```

Comment:
Similar comment to sync client

Reply:
yeah same, @PhiBrandon have you already implemented litellm support ? if yes can you share with your work